### PR TITLE
L3 default support for Network Load Balancer

### PR DESCRIPTION
**Pull request by drebes**
If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added support for `L3_DEFAULT` as `ip_protocol` for `google_compute_forwarding_rule` and `UNSPECIFIED` as `protocol` for `google_compute_region_backend_service` to support network load balancers that forward all protocols and ports.
```

### CODE REVIEW

File: mmv1/products/compute/api.yaml [hunk 1]
```diff
@@ -3697,6 +3698,7 @@ objects:
           - :AH
           - :SCTP
           - :ICMP
+          - :L3_DEFAULT
```
Comment (rileykarson) [inline mmv1/products/compute/api.yaml:3701]: Should we make the docs changes to `ports` / `allPorts` that come w/ `L3_DEFAULT` as well?

**Review by rileykarson — COMMENTED** [review]
One small question, running VCR to get test results otherwise

File: mmv1/products/compute/api.yaml (hunk 1 above)
Comment (rileykarson) [reply mmv1/products/compute/api.yaml:3701]: The changes to the descriptions outlined in the user guide, specifically!

**Review by rileykarson — APPROVED** [review]
Code LGTM (ignore the VCR failure, I superseded it w/ a manual run)

Just waiting on resolution re: descriptions

File: mmv1/products/compute/api.yaml (hunk 1 above)
Comment (drebes) [reply mmv1/products/compute/api.yaml:3701]: Done

**Comment by rileykarson** [comment]
Not sure what's up w/ the test failure, `iam.go` didn't import `sort` for no good reason. Ignoring that.

**Pull request merged**